plugins/tds40.js

```diff
@@ -1502,18 +1502,35 @@ tds = {
                     // If we have a coastline around an Island, decide if we are going make an Island
                     // or a Coastline
                     if (tags.natural == 'coastline')
-                    {
-                        if (geometryType == 'Area') // Islands are Areas
+                        if (geometryType == 'Line')
                         {
-                            delete tags.natural;
+                            attrs.F_CODE = 'BA010'; // Land/Water Boundary - Line
+                            delete tags.place;
                         }
-                        else if (geometryType =='Line') // Coastlines are lines
+                        else
                         {
-                            delete tags.place;
+                            // NOTE: Islands can be Points or Areas
+                            attrs.F_CODE = 'BA030'; // Island
+                            delete tags.natural;
                         }
-                    }
                     break;
 
+            case 'island':
+            case 'islet':
+                if (tags.natural == 'coastline')
```

Comment:
Google says to use curly braces, too: https://google.github.io/styleguide/jsguide.html